src/open_prime_hunters_rando/entities/pickup.py

```diff
@@ -1,70 +1,75 @@
-from open_prime_hunters_rando.constants import ITEM_TYPES_TO_IDS, get_entity
+from construct import Construct, Container
 
+from open_prime_hunters_rando.constants import get_entity
+from open_prime_hunters_rando.entities.entity_type import EntityType
 
-def patch_pickups(entity_file: memoryview, pickups: list) -> None:
+
+def patch_pickups(entity_file: Construct, pickups: list) -> None:
     for pickup in pickups:
         entity_id = pickup["entity_id"]
         new_entity_type = pickup["entity_type"]
+        entity_idx = get_entity(entity_file, entity_id)
 
-        data_offset, old_entity_type = get_entity(entity_file, entity_id)
-
-        # Update the header to use the new item type
-        entity_file[data_offset] = new_entity_type
+        entity = entity_file.entities[entity_idx]
+        header = entity.data.header
 
-        # Entity Data is offset 0x28 (40) from the start of the header
-        main_data = data_offset + 40
+        old_entity_data = entity.data
+        old_entity_type = old_entity_data.header.entity_type
 
         # Update ItemSpawn entities
         # Entity was ItemSpawn
-        if old_entity_type == 4:
+        if old_entity_type == EntityType.ITEM_SPAWN:
             # Entity is still ItemSpawn
-            if new_entity_type == old_entity_type:
-                entity_file[main_data + 4] = ITEM_TYPES_TO_IDS[pickup["item_type"]]
+            if new_entity_type == EntityType.ITEM_SPAWN.value:
```

Comment:
I don't think the `.value` is necessary?

Reply:
Not using the value is comparing `4` to <EntityType.ItemSpawn: 4>

Reply:
ah. in that case, instead of `new_entity_type = pickup["entity_type"]`, do `new_entity_type = EntityType(pickup["entity_type"])` and then make the change removing `.value`